### PR TITLE
fix: improved hdr brightness fingerprint compatibility

### PR DESCRIPTION
**Pull request by OxrxL**

### CODE REVIEW

**Comment by OxrxL** [comment]
Tried with 17.33.42 and latest 17.36.37.

File: src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/patch/HDRBrightnessPatch.kt [hunk 1]
```diff
@@ -40,7 +41,10 @@ class HDRBrightnessPatch : BytecodePatch(
             )
         )
 
-        val method = HDRBrightnessFingerprint.result!!.mutableMethod
+        var result = HDRBrightnessFingerprint.result
+        if (result == null) result = HDRBrightnessLegacyFingerprint.result
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/patch/HDRBrightnessPatch.kt:45]: Is it possible to only rely on a single fingerprint which works for both versions?

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/patch/HDRBrightnessPatch.kt (hunk 1 above)
Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/patch/HDRBrightnessPatch.kt:45]: Unfortunately no. On versions prior to 17.33.42 the structure of the method changes completely.

File: src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/fingerprints/HDRBrightnessFingerprint.kt [hunk 2]
```diff
@@ -14,5 +18,18 @@ import app.revanced.patches.youtube.misc.hdrbrightness.annotations.HDRBrightness
 @Version("0.0.1")
 object HDRBrightnessFingerprint : MethodFingerprint(
     "V",
-    strings = listOf("c.SettingNotFound;", "screen_brightness", "android.mediaview", "lux"),
+    customFingerprint = { methodDef ->
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/hdrbrightness/fingerprints/HDRBrightnessFingerprint.kt:21]: The custom fingerprint is very complex, instead a parent fingerprint should be used as previously mentioned.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by OxrxL** [comment]
Fixed to reduce the unnecessary wide compatibility (which now it start from stable version 17.33.42)

**Pull request merged**